test/shared/src/main/scala/zio/test/Assertion.scala

```diff
@@ -110,6 +110,17 @@ object Assertion extends AssertionVariants {
     final case class Function(name: String, paramLists: List[List[RenderParam]]) extends Render
     final case class Infix(left: RenderParam, op: String, right: RenderParam)    extends Render
 
+    /**
+     * Creates a string representation of a class name.
+     */
+    def className[A](C: ClassTag[A]): String =
+      try {
+        C.runtimeClass.getSimpleName
+      } catch {
+        case t: InternalError if t.getMessage == "Malformed class name" =>
```

Comment:
It may be worth mentioning the Scala bug here, as this code can probably be dropped at some point in the future.

Reply:
Good idea. Done now.